mod_ci/controllers.py

```diff
@@ -866,6 +868,56 @@ def show_maintenance():
     }
 
 
+@mod_ci.route('/blocked_users')
+@login_required
+@check_access_rights([Role.admin])
+@template_renderer()
+def blocked_users():
+        blocked_users = BlockedUsers.query.order_by(BlockedUsers.userID)
+
+        usernames = {}  # Initialize usernames dictionary
+        usernames = {u.userID: 'Error, cannot get username' for u in blocked_users}
+        for key in usernames.keys():
+            # Fetch usernames from GitHub API
+            try:
+                api_url = requests.get('https://api.github.com/user/{}'.format(key), timeout=10)
+            except requests.exceptions.RequestException:
+                break
+            userdata = api_url.json()
```

Comment:
This'll need to be inside the try-except, or the api_url will be undefined if an exception occurs.

Same goes for next line.